pyccel/parser/semantic.py

```diff
@@ -1896,7 +1896,7 @@ def _get_indexed_type(self, base, args, expr):
                 raise errors.report(f"Unknown annotation base {base}\n"+PYCCEL_RESTRICTION_TODO,
                         severity='fatal', symbol=expr)
             rank = 1
-            if len(args) == 2 and args[1] is LiteralEllipsis():
+            if (len(args) == 2 and args[1] is LiteralEllipsis()) or (len(args) > 0):
```

Comment:
if `len(args) == 2 and args[1] is LiteralEllipsis()` is true, then `len(args) > 0` is also true. Are you sure you need both conditions?

Reply:
In the case of `int : list[int]`, the `args` holds `int`, so we need to check for it.

Reply:
I agree. But that doesn't mean that you need both conditions

Reply:
If it's okay with you, I will remove the first condition. However, I think it has a value in the code, and honestly, I don't know its benefits.

Reply:
The reason for specifying `len(args) == 2 and args[1] is LiteralEllipsis()` was because the code below was only designed to handle expressions like `tuple[int, ...]` now that it is going to be more general I think it makes sense to change the condition. There is certainly no value in keeping slower code that evaluates more things but gives the same result. I am fine to remove the condition :+1:

Reply:
Fixed : d0bd13c